File: backend/src/main/java/com/springbootjwtpostgres/backend/product/Product.java
package com.springbootjwtpostgres.backend.product;

import com.springbootjwtpostgres.backend.basemodels.BaseEntity;
import com.springbootjwtpostgres.backend.inventory.Inventory;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;


@Entity
@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
public class Product extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String productName;
    private String productCode;
    private double productPriceIn;
    private double productPriceOut;
    private String note;
    private Date createdAt;

    @OneToOne(optional = false)
    private Inventory inventory;
}

File: frontend/src/app/models/product/Product.ts
import { BaseEntity } from '../base/BaseEntity';
import { Inventory } from '../inventory/Inventory';

export class Product extends BaseEntity {
  productName!: string;
  productCode?: string;
  productPriceIn?: number;
  productPriceOut?: number;
  note?: string;
  inventory?: Inventory;
}
